src/openfermion/ops/_interaction_tensor.py

```diff
@@ -84,103 +85,114 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
+                n-body interactions. For instance, n_body_tensors[2] is a
+                n_qubits x n_qubits x n_qubits x n_qubits numpy array of
+                complex numbers.
         """
-        if constant is None:
-            constant = 0.0
-        self.constant = constant
-        self.one_body_tensor = one_body_tensor
-        self.two_body_tensor = two_body_tensor
-        self.n_qubits = self.one_body_tensor.shape[0]
+        # initialize constant
+        if constant is not None:
+            self.constant = constant
+        else:
+            self.constant = 0.
+
+        # initialize n_body_tensors
+        if args:
+            # tensors were passed in as arguments
+            self.n_body_tensors = dict()
+            for i in range(len(args)):
+                self.n_body_tensors[i + 1] = args[i]
+        elif n_body_tensors:
+            # tensors were passed in directly as dictionary
+            self.n_body_tensors = n_body_tensors
+        else:
+            # no tensors were given
+            self.n_body_tensors = dict()
+
+        # initialize n_qubits
+        if n_qubits:
+            self.n_qubits = n_qubits
+        elif 1 in self.n_body_tensors:
+            self.n_qubits = self.n_body_tensors[1].shape[0]
+        else:
+            raise ValueError("Could not determine n_qubits.")
 
     def __getitem__(self, args):
         """Look up matrix element.
 
         Args:
-            Ints giving indices of tensor. Either p,q or p,q,r,s.
+            Ints giving indices of tensor. Should have even length.
 
         Raises:
-            ValueError: args must be of length 2 or 4.
-            ValueError: args must be of length 0, 2 or 4.
+            ValueError: args must be of even length.
         """
-        if len(args) == 4:
-            p, q, r, s = args
-            return self.two_body_tensor[p, q, r, s]
-        elif len(args) == 2:
-            p, q = args
-            return self.one_body_tensor[p, q]
-        elif not len(args):
+        args = tuple(args)
+        if not len(args):
             return self.constant
+        elif len(args) % 2 == 0:
+            return self.n_body_tensors[len(args) // 2][args]
         else:
-            raise ValueError('args must be of length 0, 2, or 4.')
+            raise ValueError('args must be of even length.')
 
     def __setitem__(self, args, value):
         """Set matrix element.
 
         Args:
-            Ints giving indices of tensor. Either p,q or p,q,r,s.
+            Ints giving indices of tensor.
 
         Raises:
-            ValueError: args must be of length 2 or 4.
+            ValueError: args must be of even length.
         """
-        if len(args) == 4:
-            p, q, r, s = args
-            self.two_body_tensor[p, q, r, s] = value
-        elif len(args) == 2:
-            p, q = args
-            self.one_body_tensor[p, q] = value
-        elif not len(args):
+        args = tuple(args)
+        if not len(args):
```

Comment:
None value and empty list + dicts will evaluate to length 1.  Are you sure you want this behavior to return the constant?

Reply:
I don't think I understand. If `a` is an empty list or an empty dictionary, then `len(a)` evaluates to `0`. `len(None)` returns an error.

Reply:
I agree with Kevin